src/server/scripts/Pandaria/GateoftheSettingSun/instance_gate_setting_sun.cpp

```diff
@@ -0,0 +1,399 @@
+        {}
+
+        void Initialize()
+        {
+            SetBossNumber(EncounterCount);
+            LoadDoorData(doorData);
+
+            kiptilakGuid            = 0;
+            gadokGuid               = 0;
+            rimokGuid               = 0;
+            raigonnGuid             = 0;
+            raigonWeakGuid          = 0;
+            
+            firstDoorGuid           = 0;
+
+            cinematicTimer          = 0;
+            cinematicEventProgress  = 0;
+
+            wallCGuid               = 0;
+            portalTempGadokGuid     = 0;
+
+            memset(dataStorage, 0, MAX_DATA * sizeof(uint32));
+
+            bombarderGuids.clear();
+            bombStalkerGuids.clear();
+            mantidBombsGUIDs.clear();
+            rimokAddGenetarorsGUIDs.clear();
+            artilleryGUIDs.clear();
+            secondaryDoorGUIDs.clear();
+        }
+
+        void OnDestroy(InstanceMap* pMap)
+        {
+            if (Creature* weakSpot = instance->GetCreature(GetObjectGuid(NPC_WEAK_SPOT)))
+                weakSpot->_ExitVehicle();
+        }
+
+        void OnPlayerEnter(Player* player)
+        {
+            if (GetData(DATA_BRASIER_CLICKED) == NOT_STARTED)
+                player->SetPhaseMask(1, true);
+            else
+                player->SetPhaseMask(2, true);
+        }
+
+        void OnPlayerLeave(Player* player)
+        {
+            player->SetPhaseMask(1, true);
+        }
+
+        void OnCreatureCreate(Creature* creature)
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_KIPTILAK:          kiptilakGuid    = creature->GetGUID().GetEntry();                  return;
+                case NPC_GADOK:             gadokGuid       = creature->GetGUID().GetEntry();                  return;
+                case NPC_RIMOK:             rimokGuid       = creature->GetGUID().GetEntry();                  return;
+                case NPC_RAIGONN:           raigonnGuid     = creature->GetGUID().GetEntry();                  return;
+                case NPC_KRITHUK_BOMBARDER: bombarderGuids.push_back(creature->GetGUID().GetEntry());          return;
+                case NPC_BOMB_STALKER:      bombStalkerGuids.push_back(creature->GetGUID().GetEntry());        return;
+                case NPC_ADD_GENERATOR:     rimokAddGenetarorsGUIDs.push_back(creature->GetGUID().GetEntry()); return;
+                case NPC_ARTILLERY:         artilleryGUIDs.push_back(creature->GetGUID().GetEntry());          return;
+                default:                                                                            return;
+            }
+        }
+
+        virtual void OnCreatureRemove(Creature* creature)
+        {
+            switch (creature->GetEntry())
+            {
+                case NPC_KRITHUK_BOMBARDER:
+                    for (std::list<uint64>::iterator it = bombarderGuids.begin(); it != bombarderGuids.end(); ++it)

```

Comment:
as I said, GUIDs are type of ObjectGuid, also there you can use range-based for loop